server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -402,6 +403,7 @@ private void closeRepository(Repository repository) {
      */
     private Repository createRepository(RepositoryMetaData repositoryMetaData, Map<String, Repository.Factory> factories) {
         logger.debug("creating repository [{}][{}]", repositoryMetaData.type(), repositoryMetaData.name());
+        validate(repositoryMetaData.name());
```

Comment:
this is a method that is internally called both on repository creation, but also after a restart when a node rejoins a cluster with an existing repository. I think we should do the validation only for new repository creation requests, and not retroactively apply to old repositories (which will break them). This means that the validation should be done in `registerRepository`, not `createRepository`.